packages/flutter/lib/src/cupertino/route.dart

```diff
@@ -709,3 +711,125 @@ class _CupertinoEdgeShadowPainter extends BoxPainter {
     canvas.drawRect(rect, paint);
   }
 }
+
+
+class _CupertinoModalPopupRoute<T> extends PopupRoute<T> {
+  _CupertinoModalPopupRoute({
+    this.builder,
+    this.barrierLabel,
+    RouteSettings settings,
+  }) : super(settings: settings);
+
+  final WidgetBuilder builder;
+
+  @override
+  Duration get transitionDuration => const Duration(milliseconds: 500);
+
+  @override
+  bool get barrierDismissible => true;
+
+  @override
+  final String barrierLabel;
+
+  @override
+  Color get barrierColor => null; //const Color(0xFF8E8E93);
+
+  AnimationController _animationController;
+
+  @override
+  Animation<double> createAnimation() {
+    return new CurvedAnimation(
+        parent: _animationController,
+        curve: Curves.decelerate,
+    );
+  }
+
+  @override
+  AnimationController createAnimationController() {
+    assert(_animationController == null);
+    _animationController = new AnimationController(
+      duration: const Duration(milliseconds: 500),
```

Comment:
This feels too long to me, but if that's what iOS does, then it's fine.

Reply:
Yes, you are correct, was waiting to play with duration and curve until I got the positioning of the sheet correct (should have made a note of this)!